src/node_file.cc

```diff
@@ -854,6 +872,43 @@ void FromNamespacedPath(std::string* path) {
 #endif
 }
 
+static inline int GetValidMode(Environment* env,
+                               Local<Value> mode_v,
+                               std::string_view type) {
+  if (!mode_v->IsInt32() && !mode_v->IsNullOrUndefined()) {
+    THROW_ERR_INVALID_ARG_TYPE(env, "mode must be int32 or null/undefined");
+    return -1;
+  }
+
+  int min = kMinimumAccessMode;
+  int max = kMaximumAccessMode;
+  int def = F_OK;
+
+  if (type == "copyFile") {
+    min = kMinimumCopyMode;
+    max = kMaximumCopyMode;
+    def = mode_v->IsNullOrUndefined() ? kDefaultCopyMode
+                                      : mode_v.As<Int32>()->Value();
+  } else if (type != "access") {
+    THROW_ERR_INVALID_ARG_TYPE(
+        env, "type must be equal to \"copyFile\" or \"access\"");
+    return -1;
+  }
```

Comment:
Stringly-typed C++ isn't exactly wrong but it's unidiomatic. Can you use an enum instead? You can then also simply CHECK it's one or the other.

Reply:
CHECK will crash the process. Throwing a validation error is correct in here

Reply:
We don't need to gracefully validate arguments that are controlled by us.

Reply:
Yes, I misunderstood the comment here. I agree. Check is sufficient.